common/structures.go

```diff
@@ -30,7 +30,14 @@ func MakeIPv4(a, b, c, d byte) IP {
 }
 
 func IPv4ToInt(ip net.IP) uint64 {
-	return uint64(ip[12])<<24 | uint64(ip[13])<<16 | uint64(ip[14])<<8 | uint64(ip[15])
+	switch len(ip) {
+	case 4: // IPv4
```

Comment:
It's a bugfix.
If we want a refactor, then renaming, changing return type to uint32 and so on would be fair game.